tests/python_package_test/test_dask.py

```diff
@@ -805,6 +808,67 @@ def f(part):
         assert 'foo' in str(info.value)
 
 
+@pytest.mark.parametrize('task,output', tasks_outputs)
+def test_training_having_workers_without_data(client, task, output):
+
+    def repartition(collection):
+        if collection is None:
+            return
+        if isinstance(collection, da.Array):
+            return collection.rechunk(*collection.shape)
+        return collection.repartition(npartitions=1)
+
+    if task == 'ranking':
+        X, y, w, g, dX, dy, dw, dg = _create_ranking_data(
+            output=output,
+            group=None
+        )
+        dask_model_factory = lgb.DaskLGBMRanker
+        local_model_factory = lgb.LGBMRanker
+    else:
+        X, y, w, dX, dy, dw = _create_data(
+            objective=task,
+            output=output
+        )
+        g = dg = None
```

Comment:
```suggestion
        g = None
        dg = None
```

Please put these on separate lines.